ecllibrary/teststd/uni/TestRemoveSuffix.ecl

```diff
@@ -0,0 +1,88 @@
+    //Check action on a string containing different variations/combinations of numbers and other characters.
+    EXPORT Test24 := ASSERT(Uni.RemoveSuffix('123abc 23.6 abc123','23.6 abc123','') = '123abc ');
+    //Test other space characters (< 0x20).
+    EXPORT Test26 := ASSERT(Uni.RemoveSuffix('\nt\t','t','') = '\nt\t');
+    //Check action on a string containing latin diacritical marks.
+    EXPORT Test27 := ASSERT(Uni.RemoveSuffix(U'À à',U'à','') = U'À ');
+    EXPORT Test28 := ASSERT(Uni.RemoveSuffix(U'ȭ š',U'š','') = U'ȭ ');
+    //Check action on a string containing Spanish words with latin accents.
+    //Translation: "The deceased changed the girls"
+    EXPORT Test29 := ASSERT(Uni.RemoveSuffix(U'El difunto cambió las niñas',U'niñas','') = U'El difunto cambió las ');
+    //Check action on a string containing Chinese characters.
+    //Translation: "I am a computer"
+    EXPORT Test30 := ASSERT(Uni.RemoveSuffix(U'我是電腦',U'腦','') = U'我是電');
+    //Check action on a string containing Modern Greek characters.
+    //Translation: "Do you come here often?"
+    EXPORT Test31 := ASSERT(Uni.RemoveSuffix(U'Έρχεσαι συχνά εδώ;',U'εδώ;','') = U'Έρχεσαι συχνά ');
+    //Testcases 32 and 33 test for bidirectional capabilities with scripts in arabic and hebrew.
+    //Check action on arabic lettering with accent marks. Bidirectional.
+    //Translation: "Good morning"
+    EXPORT Test32 := ASSERT(Uni.RemoveSuffix(U'صباح الخير',U'الخير','') = U'صباح');
+    //Check action on hebrew lettering with accent marks (called pointing). Bidirectional.
+    //Translation: (not a phrase, 2 different words separated by a space)
+    EXPORT Test33 := ASSERT(Uni.RemoveSuffix(U'קָמָץ שִׁי״ן',U'שִׁי״ן','') = U'קָמָץ');
+    //Check action on surrogate pairs.
+    EXPORT Test34 := ASSERT(Uni.RemoveSuffix(U'x𐐀',U'x𐐀','') = U'');
+    EXPORT Test35 := ASSERT(Uni.RemoveSuffix(U'𐐀',U'𐐀x','') = U'𐐀');
+    EXPORT Test37 := ASSERT(Uni.RemoveSuffix(U'x𐐀',U'𐐀','') = U'x');
+    //Check action with normalization forms
+    EXPORT Test39 := ASSERT(Uni.RemoveSuffix(U'Ç̌',U'Ç̌','') = U'');
+    EXPORT Test40 := ASSERT(Uni.RemoveSuffix(U'Ç̌',U'Ç̌','NFC') = U'');
+    DATA r1 := x'43002703';
+    UNICODE t1 := TRANSFER(r1, UNICODE);
+    DATA r2 := x'c700';
+    UNICODE t2 := TRANSFER(r2, UNICODE);
+    EXPORT Test41 := ASSERT(Uni.RemoveSuffix(t1,t2,'NFC') = U'');
+    DATA r1 := x'43002703';
+    UNICODE t1 := TRANSFER(r1, UNICODE);
+    DATA r2 := x'c700';
+    UNICODE t2 := TRANSFER(r2, UNICODE);
+    EXPORT Test42 := ASSERT(Uni.RemoveSuffix(t1,t2,'NFD') = U'');
+    DATA r1 := x'43002703';
+    UNICODE t1 := TRANSFER(r1, UNICODE);
+    DATA r2 := x'c700';
+    UNICODE t2 := TRANSFER(r2, UNICODE);
+    EXPORT Test43 := ASSERT(Uni.RemoveSuffix(t1,t2,'NFKC') = U'');
+    DATA r1 := x'43002703';
+    UNICODE t1 := TRANSFER(r1, UNICODE);
+    DATA r2 := x'c700';
+    UNICODE t2 := TRANSFER(r2, UNICODE);
+    EXPORT Test44 := ASSERT(Uni.RemoveSuffix(t1,t2,'NFKD') = U'');
```

Comment:
Similar to the startsWith PR, it would be good to have some examples which only pass with some normalizations.   Also, best if it doesn't match the entire string, so the result isn't U''.